src/algebra/big_operators.lean

```diff
@@ -752,6 +752,77 @@ lemma prod_update_of_mem [decidable_eq α] {s : finset α} {i : α} (h : i ∈ s
   (∏ x in s, function.update f i b x) = b * (∏ x in s \ (singleton i), f x) :=
 by { rw [update_eq_piecewise, prod_piecewise], simp [h] }
 
+/-- If a product of a `finset` of size at most 1 is 1, so are the
+terms in that product. -/
+lemma eq_one_of_card_le_one_of_prod_eq_one {s : finset α} (hc : s.card ≤ 1) {f : α → β}
+    (h : ∏ x in s, f x = 1) : ∀ x ∈ s, f x = 1 :=
```

Comment:
Can't you generelize the `= 1` to `= b` for arbitrary `b`?

Reply:
Done.